### PR TITLE
Fetch Caching #48

### PR DESCRIPTION
**Pull request by alerosad**

### CODE REVIEW

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]

File: src/hooks/cache/caching.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+				var cCode = key[i];
+				if (cache[cCode]) {
+					foundItem = cache[cCode];
+				}
+			}
+		} else {
+			foundItem = cache[key];
+		}
+		if (foundItem == undefined) {
+			return 0;
```
Comment (arcepaul) [inline src/hooks/cache/caching.js:49]: Do we want to return `0` here?

File: src/hooks/cache/caching.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { useReducer } from 'react';
+import { setCaching } from './cacheActions';
+
+function reducer(state, action) {
+	switch (action.set_cache.type) {
+		case 'SET_CACHE':
+			if (Array.isArray(action.key) && action.key.length > 1) {
```
[the remark this answers is not in the document]
Comment (bryanpizzillo) [reply src/hooks/cache/caching.js:7]: This should only be const, not even let as it should not need to be modified

File: src/hooks/cache/caching.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				var cidCache = action.key.reduce((ac, cid) => {
+					return {
+						...ac,
+						[cid]: action.payload,
+					};
+				}, {});
+				var purlCache = action.payload[0].prettyUrlName
```
Comment (seyilonge-nci) [inline src/hooks/cache/caching.js:14]: Why is there any reference to `purl` or `prettyUrlName` in this file?

File: src/hooks/cache/caching.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+export const useCache = () => {
+	const [cache, dispatch] = useReducer(reducer, {});
+
+	const getCacheItem = (key) => {
+		var foundItem;
+		//checks to see if the key is an array of c-codes
+		if (Array.isArray(key) && key.length > 1) {
+			for (var i = 0; i < key.length; i++) {
+				var cCode = key[i];
```
Comment (seyilonge-nci) [inline src/hooks/cache/caching.js:40]: Same here with `cCode` as above.

File: src/hooks/cache/caching.js (hunk 3 above)
Comment (alerosad) [reply src/hooks/cache/caching.js:14]: my bad, forgot to update this part of the code, regarding iterating through the c-codes

**Comment by bryanpizzillo** [comment]
Closing this as it has been closed by another approach.

**Pull request closed without merging**